packages/devtools_app/lib/src/inspector/inspector_tree_controller.dart

```diff
@@ -542,6 +585,151 @@ class InspectorTreeController extends Object {
+
+      if (searchMatch.isSelected) {
+        final rowToSelect = i == currentSearchMatches.length - 1
+            ? currentSearchMatches[0]
+            : currentSearchMatches[i + 1];
+
+        onSelectRow(rowToSelect);
+        break;
+      }
+    }
+  }
+
+  void dispose() {
+    if (_searchDebounce?.isActive ?? false) {
+      _searchDebounce.cancel();
+    }
+  }
+
+  Future _searchOperation;
+  Timer _searchDebounce;
+
+  @override
+  List<InspectorTreeRow> matchesForSearch(String search) {
+    updateMatches([]);
+    searchInProgress = true;
+
+    if (_searchDebounce?.isActive ?? false) {
+      _searchDebounce.cancel();
+    }
+    _searchDebounce =
+        Timer(Duration(milliseconds: search.isEmpty ? 0 : 300), () async {
+      // do something with query
+
+      // Abort any ongoing search operations and start a new one
+      if (_searchOperation != null) {
+        try {
+          await _searchOperation;
+        } catch (e) {
+          log(e, LogLevel.error);
+        }
+      }
+      searchInProgress = true;
+
+      // Start new search operation
+      _searchOperation = matchesForSearchAsyncOld(search, _searchTarget);
+
+      await _searchOperation;
+      searchInProgress = false;
+    });
+    return [];
+  }
+
+  Future<void> matchesForSearchAsyncOld(
+      String search, SearchTargetType searchTarget) async {
+    final matches = <InspectorTreeRow>[];
+
+    bool _firstSearchMatch = true;
+    void _updateSearchMatches() {
+      updateMatches(matches);
+
+      if (matches.isNotEmpty && _firstSearchMatch) {
+        _firstSearchMatch = false;
+        // Always select the first search match
+        onSelectRow(matches[0]);
+      }
+    }
+
+    int _statsSearchOps = 0;
+    int _statsWidgets = 0;
+
+    if (search == null ||
+        search.isEmpty ||
+        serviceManager.inspectorService == null ||
+        serviceManager.inspectorService.isDisposed) {
+      _updateSearchMatches();
+      debugPrint('Search completed, no search');
+      return;
+    }
+
+    debugPrint('Search started: ' + _searchTarget.toString());
+    final caseInsensitiveSearch = search.toLowerCase();
+
+    // Reset search matches
+    for (final row in cachedRows) {
+      _statsWidgets++;
+      setSearchMatch(row.node, false);
+    }
+    _updateSearchMatches();
+
+    for (final row in cachedRows) {
+      final diagnostic = row.node.diagnostic;
+      if (row.node == null || diagnostic == null) continue;
+
+      // Widget search begin
+      if (searchTarget == SearchTargetType.widget) {
+        final description = diagnostic.toStringShort();
+        final textPreview = diagnostic.json['textPreview'];
+
+        final searchValue = textPreview is String
+            ? description + ' ' + textPreview.replaceAll('\n', ' ')
+            : description;
+
+        _statsSearchOps++;
+        if (searchValue.toLowerCase().contains(caseInsensitiveSearch)) {
+          matches.add(row);
+          setSearchMatch(row.node, true);
```

Comment:
same comment as above. SearchControllerMixin._updateMatches should already handle this